libioc/Jail.py

```diff
@@ -429,114 +428,50 @@ def start(
         self.require_jail_stopped()
         self.require_jail_match_hostid()
 
-        try:
-            yield from self.config["resolver"].apply(
-                jail=self,
-                event_scope=event_scope
-            )
-        except Exception as e:
-            raise e
-
-        events: typing.Any = libioc.events
-        jailLaunchEvent = events.JailLaunch(jail=self, scope=event_scope)
+        jailStartEvent = libioc.events.JailStart(
+            jail=self,
+            scope=event_scope
+        )
+        yield jailStartEvent.begin()
 
+        # Start Dependant Jails
         dependant_jails_started: typing.List[JailGenerator] = []
         if start_dependant_jails is True:
             dependant_jails_seen.append(self)
             DependantsStartEvent = libioc.events.JailDependantsStart
             for event in self._start_dependant_jails(
                 self.config["depends"],
-                event_scope=event_scope,
+                event_scope=jailStartEvent.scope,
```

Comment:
It's not new. By passing an event_scope, the parent can be defined. Visually this causes the event printer of the CLI to nest them and internally they belong to a group that fails together.

Btw. I'm thinking of a way to render the stacks of events that the generator functions yield, so that they can be put in the Wiki or Handbook.